### PR TITLE
Bodybag and buckling layer fixes

### PR DESCRIPTION
**Pull request by AlaunusLux**
This PR is mostly about fixing body/cryo bag layering issues. 

Currently live and annoying to try and treat around:
image

Currently, bags are are below roller beds when first deployed. If buckled and unbuckled, they are then above, because of the + 0.1 on buckling.

If buckled on a roller bed, dragged, and then unbuckled, they are then above player mobs, because of the `layer + 1` on roller bed `Move()` proc, which also doesn't get reset. I removed this +1 on Move() because it was older than the +0.1 and seems to serve the same purpose.

The +0.1 did not have a method of reverting to the original buckled atom's layer, so I added a var and check to handle that.

I also changed bullet casing's layer to `BELOW_TABLE_LAYER` with the intent of not having them on top of stasis bags and roller beds, because it causes issues as someone trying to treat wounded after a firefight. 

Changing bullet casing's layer does have a consequence of putting bullets underneath, for example, the soil in the public garden. Putting the soil on `BELOW_TABLE_LAYER` would fix this, but with the consequence of having the bushes be on top, making clicking the soil difficult for farming purposes. Adding bushes to that layer causes consequences with side window layering (like on deck three outside the public lounge.) 

For this PR I only adjusted the bullet casing layer. I'm sure it has other unintended layering interactions. Let me know if it would be better to leave them where they are or on some other layer.

Current changes in PR: Bullets under soil/roller beds/stasis bags. 
image

(Example of why making bullets on top of soil is difficult) Screenshot shows both bullets and the soil being `BELOW_TABLE_LAYER` with bushes on top, as a reference. PR Currently has bullets underneath soil.
image

### CODE REVIEW

**Comment by AlaunusLux** [comment]
!review

**Comment by AlaunusLux** [comment]
!bugfix

**Comment by AlaunusLux** [comment]
The buckle changes may also fix boxes layering above people, as most buckle items are 2.9 and boxes are 3.

File: code/game/objects/buckling.dm [hunk 1]
```diff
@@ -63,6 +64,9 @@
 	if(MA && MA.buckled_to == src)
 		. = MA
 		MA.buckled_to = null
+		if(buckled_original_layer)
+			MA.layer = buckled_original_layer
```
Comment (AlaunusLux) [inline code/game/objects/buckling.dm:68]: Should I change this to `MA.layer = initial(MA.layer)` and get rid of the variable and check? Mostly unsure of other things that may change the buckled atom's layer that we may want to preserve.

File: code/__DEFINES/layers.dm [hunk 2]
```diff
@@ -85,6 +85,7 @@
 	#define ABOVE_WINDOW_LAYER 2.78
 	#define BELOW_OBJ_LAYER 2.89
 	#define STRUCTURE_LAYER 2.9
+	#define BODY_BAG_LAYER 2.91
```
Comment (alsoandanswer) [inline code/__DEFINES/layers.dm:88]:
```suggestion
	#define ABOVE_STRUCTURE_LAYER 2.91
```

File: code/game/objects/buckling.dm (hunk 1 above)
Comment (alsoandanswer) [reply code/game/objects/buckling.dm:68]: leave it

**Review by alsoandanswer — APPROVED** [review]

File: code/game/objects/objs.dm [hunk 3]
```diff
@@ -49,6 +49,7 @@
 	var/buckle_lying = -1 //bed-like behavior, forces mob.lying = buckle_lying if != -1
 	var/buckle_require_restraints = 0 //require people to be handcuffed before being able to buckle. eg: pipes
 	var/atom/movable/buckled = null
+	var/buckled_original_layer = null
```
Comment (NonQueueingMatt) [inline code/game/objects/objs.dm:52]: dmdoc this variable

**Review by NonQueueingMatt — CHANGES_REQUESTED** [review]

**Pull request merged**